OSDOCS-6534: adding CSI snapshotting (#62557)
openshift/openshift-docs: modules/microshift-storage-about-vol-snapshots.adoc

// Module included in the following assemblies:
//
// microshift/creating-volume-snapshots-microshift.adoc

:_content-type: CONCEPT
[id="about-volume-snapshots_{context}"]
= About volume snapshots

> Volume snapshots can be used with LVM thin volumes to help protect against data loss in {product-title}. Only the `microshift csi` container storage interface (CSI) is supported.

Reviewer: May be this line "Only the `microshift csi` container storage interface (CSI) is supported.", can be rephrased to "Only supported on Microshift default  LVMS container storage interface (CSI)"

> Volume snapshots can be used with LVM thin volumes to help protect against data loss from applications running in a {product-title} cluster. {product-title} only supports the LVMS CSI provider.

Reviewer 2: Abbreviations need to be spelled out the first time they are used in a module.

Reviewer 3: Consider wording to what the user can do with it

```suggestion
You can use volume snapshots with LVM thin volumes to help protect against data loss from applications running in a {product-title} cluster. {product-title} only supports the LVMS CSI provider.
```

[NOTE]
====

> LVMS only supports the `WaitForFirstConsumer` `volumeBindingMode`, which means the storage volume is not provisioned until a pod is ready to mount it.

Author: https://docs.openshift.com/container-platform/4.12/storage/persistent_storage/persistent_storage_local/persistent-storage-using-lvms.html#lvms-provisioning-storage-using-lvms_logical-volume-manager-storage 
I think this one is what it is, but I did restructure to follow the way it was written here. A little better.

====

.Example workload that deploys a single pod and PVC

> [source, terminal]

Reviewer 3: Throughout PR, shouldn't be a space (for consistency reasons across the repo)
```suggestion
[source,terminal]
```

----
oc apply -f - <<EOF
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: test-claim-thin
spec:
  accessModes:
  - ReadWriteOnce
  resources:
    requests:
      storage: 1Gi
  storageClassName: topolvm-provisioner-thin
---
apiVersion: v1
kind: Pod
metadata:
  name: base
spec:
  containers:

>   - command:

Reviewer 4: This command does not look right. Surely there's an entrypoint in the container image to start the web server? @copejon can you help with that?

    - sh
    - -c
    - sleep 1d

>     image: nginxinc/nginx-unprivileged:latest

Reviewer 4: We should find a safe version of this image on quay.io or registry.redhat.com to use.

>     image: nginxinc/nginx-unprivileged:latest

Author: @copejon Is a rhel8 version OK for 4.14?
